api/types/src/convert.rs

```diff
@@ -272,7 +273,27 @@ impl<'a, S: StateView> MoveConverter<'a, S> {
     ) -> Result<TransactionPayload> {
         use aptos_types::transaction::TransactionPayload::*;
         let ret = match payload {
-            Script(s) => TransactionPayload::ScriptPayload(s.try_into()?),
+            // TODO: Annotate the raw transaction argument using value annotator.
+            Script(s) => {
+                let (code, ty_args, args) = s.into_inner();
+                let func_args = self.inner.view_script_arguments(&code, &args, &ty_args);
+
+                let json_args = match func_args {
```

Comment:
Can we combine this with the entry function code? Or a helper function

Reply:
The two function is a bit different in the logic due to this difference between args type (one is Vec<u8> the other one is TransactionArgument).